scp.py

```diff
@@ -445,7 +445,10 @@ def _recv_file(self, cmd):
                 # we have to make sure we don't read the final byte
                 if size - pos <= buff_size:
                     buff_size = size - pos
-                file_hdl.write(chan.recv(buff_size))
+		data = chan.recv(buff_size)
+                if not data:
+                    raise SocketTimeout("Underlying channel was closed")
```

Comment:
Raising `SocketTimeout` means that the `except` block will be executed. Is it safe to call `chan.close()` if the channel is already closed?

Reply:
Your'e right. Looks like paramiko handles this case, but to be on the safe side I changed the exception to generic `SCPException`